lib/spack/spack/patch.py

```diff
@@ -90,7 +91,8 @@ class FilePatch(Patch):
     def __init__(self, pkg, path_or_url, level):
         super(FilePatch, self).__init__(pkg, path_or_url, level)
 
-        pkg_dir = spack.repo.dirname_for_package_name(pkg.name)
+        m = inspect.getmodule(pkg)
```

Comment:
> The commit cf65636 appears to contain all changes which account for fixing #4236

Yes, there was even a comment above:

> (In case you don't agree with the PR, I think the last commit here can be cherry-picked seamlessly to solve #4236)

So: should I move this to another PR and fix `spack.repo.dirname_for_package` to take a `Package` instead of a string?